src/pluto/operation/crop.cr

```diff
@@ -0,0 +1,31 @@
+module Pluto::Operation::Crop
+  def crop(x : Int32, y : Int32, new_width : Int32, new_height : Int32) : self
+    clone.crop!(x, y, new_width, new_height)
+  end
+
+  def crop!(x : Int32, y : Int32, new_width : Int32, new_height : Int32) : self
+    raise "Crop dimensions extend #{x + new_width - width} pixels beyond width of the image (#{width})" if (x + new_width) > width
+    raise "Crop dimensions extend #{y + new_height - height} pixels beyond height of the image (#{height})" if (y + new_height) > height
```

Comment:
An alternative approach would be to black out the pixels that extend outside of the image, rather than only allowing crops inside the image. I opted for this one because less edge cases to consider, but could be swayed the other way if desired.